### ISSUE TITLE
[CI] ExistsQueryBuilder testToQuery fails occasionally

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#55950]
https://gradle-enterprise.elastic.co/s/xzne2zm3o4ujs

Reproducible locally on 7.x with
```
./gradlew ':server:test' --tests "org.elasticsearch.index.query.ExistsQueryBuilderTests.testToQuery" -Dtests.seed=D22C32FE2BD8ABE1 -Dtests.security.manager=true -Dtests.locale=zh-TW -Dtests.timezone=America/Atka -Dcompiler.java=14 -Druntime.java=8
```

Error:
```
java.lang.AssertionError:
    Expected: an instance of org.apache.lucene.search.ConstantScoreQuery
         but: <MatchNoDocsQuery("User requested "match_none" query.")> is a org.apache.lucene.search.MatchNoDocsQuery
        at __randomizedtesting.SeedInfo.seed([D22C32FE2BD8ABE1:25D730C05A5B6E0B]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.query.ExistsQueryBuilderTests.doAssertLuceneQuery(ExistsQueryBuilderTests.java:75)
```

**Comment by cbuescher** [comment]
Seems to be a particular test seed only, so not muting

**Comment by cbuescher** [comment]
Closed by #56006
